expression/builtin_time.go

```diff
@@ -638,10 +638,28 @@ func builtinNow(args []types.Datum, ctx context.Context) (d types.Datum, err err
 		}
 	}
 
-	t, err := convertTimeToMysqlTime(time.Now(), fsp)
+	// This would consider "timestamp" session variable.
+	now, err := getSystemTimestamp(ctx)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	tr, err := types.RoundFrac(now, int(fsp))
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	t := types.Time{
+		Type: mysql.TypeTimestamp,
+		Time: types.FromGoTime(tr),
+		Fsp:  fsp,
+	}
```

Comment:
I try move line 662 to here, the test is passed. So I think we can use `convertTimeToMysqlTime` instead of line646-655.